File: CBHTTP/WebSocket/WebConnectionState.swift
// Copyright (c) 2017-2019 Coinbase Inc. See LICENSE

import Foundation

/// Represents connection state of any web connection
public enum WebConnectionState {
    /// The connection is not currently live
    case disconnected(Error?)

    /// The connection is currently live
    case connected

    /// Determine whether connection is live
    public var isConnected: Bool {
        guard case .connected = self else { return false }

        return true
    }
}

File: android/http/src/main/java/com/coinbase/wallet/http/models/WebConnectionState.kt
package com.coinbase.wallet.http.models

// Represents connection state of any web connection
sealed class WebConnectionState {
    // The connection is currently live
    class Connected : WebConnectionState()

    // The connection is not currently live
    data class Disconnected(val t: Throwable?) : WebConnectionState()

    // Determine whether web connection is established
    val isConnected: Boolean
    get() = when (this) {
        is Connected -> true
        is Disconnected -> false
    }
}
